src/Service/HealthCheck/HttpUtilities.cs

```diff
@@ -144,17 +156,34 @@ public void ConfigureApiRoute(HttpContext httpContext)
                 string databaseObjectName = entity.Source.Object;
 
                 // In case any primitive column names are present, execute the query
-                if (columnNames.Any() && entity?.Health != null)
+                if (columnNames.Any())
                 {
                     using (HttpClient client = CreateClient(apiRoute))
                     {
                         string jsonPayload = Utilities.CreateHttpGraphQLQuery(databaseObjectName, columnNames, entity.EntityFirst);
                         HttpContent content = new StringContent(jsonPayload, Encoding.UTF8, Utilities.JSON_CONTENT_TYPE);
-                        HttpResponseMessage response = client.PostAsync(apiRoute, content).Result;
+
+                        HttpRequestMessage message = new(method: HttpMethod.Post, requestUri: apiRoute)
+                        {
+                            Content = content
+                        };
+
+                        if (!(string.IsNullOrEmpty(incomingRoleToken) && string.IsNullOrEmpty(incomingRoleHeader)))
```

Comment:
Can we have this if condition separately for each header - since they are unrelated?

Reply:
also, can the header be added with the value = an empty string? because with the current condition, even if one of these tokens are empty and the other not empty, the headers will be added.